actions/executeTransaction.ts

```diff
@@ -36,15 +56,132 @@ export const executeTransaction = async (
     [instruction.account.getSingleInstruction()]
   )
 
-  const transaction = new Transaction()
+  // Create proposal instruction
+  const transaction = new Transaction().add(...instructions)
+
+  console.log(instructionOption)
+  // Send transaction based on its execution option
+  switch (instructionOption) {
+    case InstructionOptions.castleRefresh:
+      return await executeWithRefresh(
+        transaction,
+        connection,
+        wallet,
+        instruction
+      )
+    case InstructionOptions.castleReconcileRefresh:
+      return await executeWithRefreshAndReconcile(
+        transaction,
+        connection,
+        wallet,
+        instruction
+      )
+    default:
+      await sendTransaction({
+        transaction,
+        wallet,
+        connection,
+        signers,
+        sendingMessage: 'Executing instruction',
+        successMessage: 'Execution finalized',
+      })
+  }
+}
+
+/**
+ * Signs and sends proposal transaction with refresh transaction adjacent to it, aiming
+ * for both to be in the same slot.
+ * @param tx Proposal transaction
+ * @param connection
+ * @param wallet
+ */
+const executeWithRefresh = async (
```

Comment:
This function has a generic name `executeWithRefresh()` but it's not a generic function because it's specific to `CastleRefreshInstruction`

Ideally `CastleRefreshInstruction` should be passed in `InstructionOptions` as execution strategy requiring to execute the instruction from a proposal together with other custom instruction and everything in this file `executeTransaction.ts` should remain generic without any Castle specific logic.

For example in the definition of the `Castle Deposit` instruction we could add `Adjacent` instructions, which would be passed here as instruction options
It would work like the `Prerequisite` instructions but they would be executed when the proposal is executed and not created

Reply:
Addressed with https://github.com/solana-labs/governance-ui/pull/661/commits/7db7abc6c63b4c27889e503b4d18515be69b0c2e

* Removed all Castle-specific code from the executeTransaction.ts file.
    * added optional parameters for preExecutionTransactions and adjacentTransaction, added documentation
    * added handling these additional parameters
    * moved Castle-specific logic out into a component